src/inspector_agent.cc

```diff
@@ -112,41 +158,55 @@ void SendTargentsListResponse(inspector_socket_t* socket, int port) {
       "  \"id\": \"%d\","
       "  \"title\": \"%s\","
       "  \"type\": \"node\","
+      "  \"url\": \"file://%s\","
       "  \"webSocketDebuggerUrl\": \"ws://localhost:%d%s\""
       "} ]";
-  char buffer[sizeof(LIST_RESPONSE_TEMPLATE) + 4096];
-  char title[2048];  // uv_get_process_title trims the title if too long
-  int err = uv_get_process_title(title, sizeof(title));
-  if (err != 0) {
-    snprintf(title, sizeof(title), "Node.js");
-  }
-  char* c = title;
-  while (*c != '\0') {
-    if (*c < ' ' || *c == '\"') {
-      *c = '_';
+  size_t buf_len = sizeof(LIST_RESPONSE_TEMPLATE) + sizeof(DEVTOOLS_HASH) +
+                   sizeof(DEVTOOLS_PATH) * 2 + 200 + path.length() * 2;
+  char* buffer = new char[buf_len];
+  size_t title_start = path.rfind('/');
+  if (title_start == std::string::npos) {
+    title_start = path.rfind('\\');
+    if (title_start == std::string::npos) {
+      title_start = 0;
     }
-    c++;
   }
-  size_t len = snprintf(buffer, sizeof(buffer), LIST_RESPONSE_TEMPLATE,
+
+  if (title_start > 0) {
+    title_start += 1;  // Skip separator
+  }
+
+  std::string title = path.substr(title_start);
+
+  for (size_t i = 0; i < title.length(); i++) {
+    const char c = title[i];
+    title[i] = (c < ' ' || c == '\"') ? '_' : c;
```

Comment:
I do not see a URL encoder readily available to use in C++ code and I do not think this URL field is important enough to justify introducing one.
This JSON output is used for target discovery (e.g. chrome://inspect) and tools should not be relying on it (I've noticed that some tools crash when this field is missing). Tools will be getting a better list of scripts from V8 through the inspector WS protocol.

My goal right now is to produce a best-effort URL - one that is correct in most common use-cases (e.g. I tested with Cyrillic characters - they worked fine, I fully expect that other UTF-8 systems will also work).